test_mq_service.py

```diff
@@ -251,6 +255,27 @@ def test_handle_message_verify_metadata_is_not_required(self):
                 json.loads(mock_event_producer.get_write_event())["host"][key], expected_results["host"][key]
             )
 
+    def test_handle_message_verify_message_key(self):
+        expected_insights_id = str(uuid.uuid4())
+        host_id = uuid.uuid4()
+
+        host_data = {
+            "display_name": "test_host",
+            "insights_id": expected_insights_id,
+            "account": "0000001",
+            "stale_timestamp": "2019-12-16T10:10:06.754201+00:00",
+            "reporter": "test",
+        }
+        message = {"operation": "add_host", "data": host_data}
+
+        with self.app.app_context():
+            with unittest.mock.patch("app.queue.ingress.host_repository.add_host") as m:
+                m.return_value = ({"id": host_id, "insights_id": None}, AddHostResults.created)
+                mock_event_producer = mockEventProducer()
+                handle_message(json.dumps(message), mock_event_producer)
+
+                self.assertEqual(mock_event_producer.get_message_key(), m.return_value[0]["id"])
```

Comment:
If we have the _patch_ call here, we can even use directly the _host_id_ variable.
```suggestion
                self.assertEqual(mock_event_producer.get_message_key(), host_id)
```

But nothing prevents us from patch from a decorator and then the _return_value_ access would be good.

```python
    @patch(
        "app.queue.ingress.host_repository.add_host",
        return_value=({"id": uuid.uuid4(), "insights_id": None}, AddHostResults.created)
    )
    def test_handle_message_verify_message_key(self, add_host):
        ...
        with self.app.app_context():
            mock_event_producer = mockEventProducer()
            handle_message(json.dumps(message), mock_event_producer)

            self.assertEqual(mock_event_producer.get_message_key(), add_host.return_value[0]["id"])

Reply:
@Glutexo Your decorator solution looks great, so, I decided to go for it here.